modules/shared/src/test/scala/org/tessellation/json/JsonSerializerSuite.scala

```diff
@@ -0,0 +1,71 @@
+package org.tessellation.json
+
+import cats.MonadThrow
+import cats.data.NonEmptySet
+import cats.effect.{IO, Resource}
+import cats.syntax.functor._
+
+import scala.collection.immutable.{SortedMap, SortedSet}
+
+import org.tessellation.currency.schema.currency.{CurrencyIncrementalSnapshot, CurrencySnapshot, CurrencySnapshotInfo}
+import org.tessellation.kryo.KryoSerializer
+import org.tessellation.schema._
+import org.tessellation.schema.height.{Height, SubHeight}
+import org.tessellation.security.hash.{Hash, ProofsHash}
+import org.tessellation.security.hex.Hex
+import org.tessellation.security.signature.Signed
+import org.tessellation.security.signature.signature.{Signature, SignatureProof}
+import org.tessellation.shared.sharedKryoRegistrar
+
+import eu.timepit.refined.auto._
+import eu.timepit.refined.types.numeric.NonNegLong
+import weaver.MutableIOSuite
+
+object JsonSerializerSuite extends MutableIOSuite {
+
+  type Res = KryoSerializer[IO]
+
+  override def sharedResource: Resource[IO, Res] =
+    KryoSerializer.forAsync[IO](sharedKryoRegistrar)
+
+  test("should deserialize properly serialized object") { implicit res =>
+    currencyIncrementalSnapshot[IO](Hash.empty, CurrencySnapshotInfo(SortedMap.empty, SortedMap.empty)).map { signedSnapshot =>
+      val serialized = JsonSerializer.serialize(signedSnapshot)
+      val deserialized = JsonSerializer.deserialize[Signed[CurrencyIncrementalSnapshot]](serialized)
+      expect.same(Right(signedSnapshot), deserialized)
+    }
+  }
+
+  test("should not deserialize different serialized object") { implicit res =>
+    currencyIncrementalSnapshot[IO](Hash.empty, CurrencySnapshotInfo(SortedMap.empty, SortedMap.empty)).map { signedSnapshot =>
+      val serialized = JsonSerializer.serialize(signedSnapshot)
+      val deserialized = JsonSerializer.deserialize[CurrencySnapshot](serialized)
+      expect.same(true, deserialized.isLeft)
+    }
+  }
+
+  private def currencyIncrementalSnapshot[F[_]: MonadThrow: KryoSerializer](
+    hash: Hash,
+    currencySnapshotInfo: CurrencySnapshotInfo
```

Comment:
You are using this function twice with the same "empty" value for this parameter. You could remove the parameter altogether or give a default value for it if you think injecting different value will be useful in the future.